openjfx/samples/Ensemble8/build.gradle

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2018, OpenJFX and/or its affiliates.
+ * All rights reserved. Use is subject to license terms.
+ *
+ * This file is available and licensed under the following license:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the distribution.
+ *  - Neither the name of OpenJFX nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+apply plugin:'application'
+
+repositories {
+    mavenCentral()
+}
+
+mainClassName = "ensemble.EnsembleApp"
+
+def FileCollection apachecp = files(
+   "./lib/lucene-core-7.1.0.jar",
+   "./lib/lucene-grouping-7.1.0.jar",
+   "./lib/lucene-queryparser-7.1.0.jar")
+
```

Comment:
These jars somehow need to show up in the 'lib' dir or else you will need to change the logic that builds the jar. As it is, the assemble and build tasks are broken. For example:

```
$ gradle assemble
...
> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :jar FAILED

FAILURE: Build failed with an exception.

* Where:
Build file '/localhome/kcr/javafx/github/jfx-samples/openjfx/samples/Ensemble8/build.gradle' line: 131

* What went wrong:
Execution failed for task ':jar'.
> /localhome/kcr/javafx/github/jfx-samples/openjfx/samples/Ensemble8/lib does not exist.
```

Reply:
I add the libs now by taking them from the classpath. In general, I try to avoid putting external dependencies (copy-by-value) in a repository. If we need more recent versions from the dependencies, we can change the version number now without having to install new binaries in the repo.